### DISCUSSION TITLE
Adding a new disk for /nsm

### DISCUSSION THREAD

**Discussion by juddcbrown** [Security-Onion-Solutions/securityonion#3287] [Unsupported Versions]
Sorry for the re-ask.

Could you provide equivalent steps to run on 2.3.30 as on this page  https://docs.securityonion.net/en/16.04/new-disk.html?

Not sure what services and how to stop them before moving /nsm.

Thanks, Judd

**Comment by TOoSmOotH** [comment]
Easiest way to do this is to run:

`sudo systemctl disable salt-minion`
`reboot
`

That should prevent most things from starting. If it is a manager you will need to do 'service docker stop' after the reboot. 

Then do:

`sudo mv /nsm /nsm.old`
`sudo mkdir /nsm`
`add your new file system to mount to /nsm in fstab
`
`sudo mount -a`

Make sure its mounted correctly then run:

`sudo mv /nsm.old/* /nsm/`
`sudo rm -rf /nsm.old`
`sudo systemctl enable salt-minion`
`sudo reboot`

Everything should come up on the new disk after reboot.

**Reply by juddcbrown** [reply]
That's awesome, thank you!

**Reply by juddcbrown** [reply]
Close, but it seems Python is still holding up the move. See attached.


[nsm_move.docx](https://github.com/Security-Onion-Solutions/securityonion/files/6093765/nsm_move.docx)

**Reply by juddcbrown** [reply]
Could this be why?

[root@mn-c1cx-so2-search /]# mount | grep nsm
/dev/mapper/system-nsm on /nsm type xfs (rw,relatime,seclabel,attr2,inode64,noquota)

**Reply by TOoSmOotH** [reply]
If you are using LVM why not just expand the volume?

**Reply by juddcbrown** [reply]
> Easiest way to do this is to run:
> 
> `sudo systemctl disable salt-minion`
> `reboot `
> 
> That should prevent most things from starting. If it is a manager you will need to do 'service docker stop' after the reboot.
> 
> Then do:
> 
> `sudo mv /nsm /nsm.old`
> `sudo mkdir /nsm`
> `add your new file system to mount to /nsm in fstab `
> `sudo mount -a`
> 
> Make sure its mounted correctly then run:
> 
> `sudo mv /nsm.old/* /nsm/`
> `sudo rm -rf /nsm.old`
> `sudo systemctl enable salt-minion`
> `sudo reboot`
> 
> Everything should come up on the new disk after reboot.

I was wondering if this move could be accomplished with a cp instead of a move. Would I get all the data I need in nsm.old?

> `sudo **cp** /nsm /nsm.old`

**Comment by juddcbrown** [comment]
Storage will be on a RAID 6 Synology device with 32TB.

On Sat, Mar 6, 2021 at 09:32 Mike Reeves <notifications@github.com> wrote:

> If you are using LVM why not just expand the volume?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/3287#discussioncomment-436051>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH4XT7TOTAJ32X2IBI6JDE3TCJDIJANCNFSM4YTSTSLA>
> .
>
-- 
Judd Brown

**Reply by juddcbrown** [reply]
Forgot to mention that 2.3 installed LVM via the .iso. I didn't have an option one way or the other (that I recall).

Thanks for your help Mike.

**Reply by juddcbrown** [reply]
Is it possible to use cp recursively in place of mv? Perhaps cp -rp /nsm /nsm.old?

[the remark this answers is not in the document]
**Reply by juddcbrown** [reply]
There was no comment in your post. Did you have an idea for me to try?

[the remark this answers is not in the document]
**Reply by pimpernoni** [reply]
Judd,
Did you ever find a way to move the /nsm folder to a different logical drive.? I installed 2.3.90 and I couldn't see the RAID 1 volume that I had set up during the installation, just the two component physical drives. 

It was recommended to load the /nsm on the same drive that /boot was installed, then move the /nsm subsequently. 

Following this instruction: https://docs.securityonion.net/en/2.3/new-disk.html

However, when I try to mv /nsm /nsm.old, I get  "mv: cannot move ‘/nsm’ to ‘/nsm.old’: Device or resource busy".
I suspect this has to do with the LVM formatting of the /dev/mapper/system-nsm seen when I check #: mount.

How can I effect the change from the current /nsm folder to the new RAID1?

P

[the remark this answers is not in the document]
**Reply by dougburks** [reply]
@pimpernoni Looks like you created a separate discussion over at https://github.com/Security-Onion-Solutions/securityonion/discussions/6661 and that is the proper place to continue this discussion.

**Comment by juddcbrown** [comment]
Hey there,

You need to run umount /nsm before it will allow you to copy. This unmounts
the fs from the Logical Drive Manager, freeing up any locks it may have on
the fs.

Good luck!

On Mon, Dec 20, 2021 at 17:53 pimpernoni ***@***.***> wrote:

> Judd,
> Did you ever find a way to move the /nsm folder to a different logical
> drive.? I installed 2.3.90 and I couldn't see the RAID 1 volume that I had
> set up during the installation, just the two component physical drives.
>
> It was recommended to load the /nsm on the same drive that /boot was
> installed, then move the /nsm subsequently.
>
> Following this instruction:
> https://docs.securityonion.net/en/2.3/new-disk.html
>
> However, when I try to mv /nsm /nsm.old, I get "mv: cannot move ‘/nsm’ to
> ‘/nsm.old’: Device or resource busy".
> I suspect this has to do with the LVM formatting of the
> /dev/mapper/system-nsm seen when I check #: mount.
>
> How can I effect the change from the current /nsm folder to the new RAID1?
>
> P
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Security-Onion-Solutions/securityonion/discussions/3287#discussioncomment-1850108>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH4XT7TZG3Y4GSHG7XJPYO3UR66ZJANCNFSM4YTSTSLA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> <Security-Onion-Solutions/securityonion/repo-discussions/3287/comments/1850108
> @github.com>
>
-- 
Judd Brown
